src/monitors/ModuleMonitor.v3

```diff
@@ -0,0 +1,273 @@
+// Copyright 2023 Wizard Authors. All rights reserved.
+// See LICENSE for details of Apache 2.0 license.
+
+class ModuleMonitor(module: Module) {
+	// Attach a callback that is called when func (non-import) is called
+	def beforeFuncCall(func: FuncDecl, f: DynamicLoc -> Resumption) {
```

Comment:
Since these probes fire after the frame and arguments are set up (because it's done with an instruction probe in the target function), I think we should refer to this event as a "function execution" whereas a "function call" event happens in the caller, before the callee function's frame is set up. In that case, it'd be "beforeExecFunc"?

wdyt?

Reply:
Yeah, that makes sense to me.